template/query.py

```diff
@@ -21,6 +22,13 @@ def __init__(self, table):
     # Return False if record doesn't exist or is locked due to 2PL
     """
     def delete(self, key):
+        # key is the unique identifier that is user facing and should map to an internal rid
+        # 1. Find the record in our page_directory - make sure record exists
+        # 2. Update delete value to true
+        # 3. Change schema encoding to all zeros
+        # 4. Update indirection pointer to point to one update prior to the most current update 
```

Comment:
So delete is a per update function? it doesn't just flag the whole record for deletion?